File: 01.Searching/02.BinarySearch/BinarySearch.js

let binarySearch = function (arr, x) {

	let start = 0, end = arr.length - 1;

	// Iterate while start not meets end
	while (start <= end) {

		// Find the mid index
		// let mid = Math.floor((start + end) / 2);
		let mid = Math.floor(start + (end - start) / 2);
		// If element is present at mid, return True
		if (arr[mid] === x)
			return mid;
		// Else look in left or right half accordingly
		else if (arr[mid] < x)
			start = mid + 1;
		else
			end = mid - 1;
	}
	// if we reach here, then element was
	// not present
	return -100; // any INVALID value is correct.
}

// Driver code
let arr = [1, 3, 5, 7, 8, 9, 10, 11];
let x = 100;
let result = binarySearch(arr, x);
if (result === -1)
	console.log("No element found")
else console.log("Found !!!")

File: 01.Searching/02.BinarySearch/BinarySearch.java

public class BinarySearch {
    // TC : O(lgN)
    static int bs(int arr[], int target) {

        int low = 0, high = arr.length - 1;

        while (low <= high) {
            // int mid = (low + high) / 2; Potential Integer Overflow
            int mid = low + (high - low) / 2;
            if (arr[mid] == target)
                return mid;
            else if (arr[mid] < target) {
                low = mid + 1;
            } else
                high = mid - 1;
        }
        // if we reach here, then element was
        // not present
        System.out.println("Target not found");
        return 0; // dummy number.
    }

    public static void main(String[] args) {
        int arr[] = { 1, 2, 4, 5, 6, 9, 11, 13 };
        int target = 11;
        System.out.println(bs(arr, target));
    }

}